### ISSUE TITLE
MA0038 false positives when using Primary Constructors

### ISSUE THREAD

**Issue by sharpninja** [meziantou/Meziantou.Analyzer#666]
### Version of the Meziantou.Analyzer NuGet package

2.0.123

### Rule Identifier

MA0038

### Target Framework

net8.0

### C# Language version

latest

### Description

Getting false positives that methods can be made static even though properties defined in the Primary Constructor are referenced.

### Reproduction Steps

Minimal code:

```c#
public class Test(IServiceProvider Services)
{
    public TResult[] GetServiceAsArray<TResult>()
    {
        return [Services.GetService<TResult>()];
    }
}
```

**Comment by meziantou** [comment]
CA1822 is deprecated, so I'll close the issue as not planned. You should use the built-in analyzer CA1822 instead as indicated in the diagnostic message.

https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822

**Comment by sharpninja** [comment]
But why would you leave a broken analyzer in the default profile?

**Comment by meziantou** [comment]
I don't know what would be the best way to ask users to migrate to the new rule. At the moment, the new rule is suggested in the diagnostic message (visible in the error window or the build output). I don't know if there is a better way to do it.

I'll remove the code later.
